risc0/zkp/src/prove/soundness.rs

```diff
@@ -0,0 +1,292 @@
+use crate::{hal::Hal, taps::TapSet, FRI_FOLD, FRI_MIN_DEGREE, INV_RATE};
+use risc0_core::field::{baby_bear, ExtElem};
+
+/// Johnson's bound
+const M: f32 = 16.0;
+
+/// Rate
+const RHO: f32 = 1.0 / crate::INV_RATE as f32;
+
+/// η in Conjecture 8.4 [BCIKS21]
+const ETA: f32 = 0.05;
+
+pub fn proven<H: Hal>(taps: &TapSet, coeffs_size: usize) -> f32 {
+    let params = parameters::<H>(taps, coeffs_size);
+    let e_proximity_gap = params.e_proximity_gap_proven();
+
+    // α = (1 + 1/2m) * sqrt(ρ)
+    let alpha = (1.0 + 1.0 / (2.0 * M)) * RHO.sqrt();
+
+    let theta = 1.0 - alpha;
+    let l_plus = {
+        let rho_plus = (params.trace_domain_size + params.biggest_combo) / params.lde_domain_size;
+        let m_plus = 1.0 / (params.biggest_combo * (alpha / rho_plus.sqrt() - 1.0));
+        let m_plus = m_plus.ceil();
+
+        (m_plus + 0.5) / rho_plus.sqrt()
+    };
+    soundness_error::<H>(taps, coeffs_size, theta, e_proximity_gap, l_plus)
+}
+
+pub fn conjectured<H: Hal>(taps: &TapSet, coeffs_size: usize) -> f32 {
```

Comment:
there are two relevant conjectured regimes -- one based on the toy problem and one based on conjecture 8.4 and conjecture 2.3. 

at least for right now, we're moving forward using the toy problem conjecture as our cryptographic foundations, so we should be careful to clarify that this is not "our" conjectured result